prow/pubsub/subscriber/subscriber.go

```diff
@@ -351,16 +351,26 @@ func (s *Subscriber) handleProwJob(l *logrus.Entry, jh jobHandler, msg messageIn
 		return err
 	}
 
-	reportProwJobFailure := func(pj *prowapi.ProwJob, err error) {
-		pj.Status.State = prowapi.ErrorState
-		pj.Status.Description = err.Error()
+	reportProwJob := func(pj *prowapi.ProwJob, state v1.ProwJobState, err error) {
+		pj.Status.State = state
+		if err != nil {
+			pj.Status.Description = err.Error()
```

Comment:
We should have a status description in non-error cases as well.

Reply:
good idea, done